epub33/core/index.html

```diff
@@ -6743,9 +6752,10 @@ <h5>The <code>body</code> Element</h5>
 										<code>[optional]</code>
 									</dt>
 									<dd>
-										<p>The relative IRI reference [[RFC3987]] of the corresponding EPUB Content
-											Document, including a <a href="#sec-media-overlays-fragids">fragment
-												identifier</a> that references the specific content.</p>
+										<p>The <a href="https://url.spec.whatwg.org/#relative-url-with-fragment-string"
+												>relative-URL-with-fragment string</a> [[URL]] of the corresponding EPUB
+											Content Document, which MUST have a <a href="#sec-media-overlays-fragids"
+												>fragment identifier</a> that references the specific content.</p>
```

Comment:
I wonder in this case if we're squeezing too much information into a single sentence.

Would it make sense to separate what textref is for from what its value must be, so reformulate more like the epub:type attribute definition along these lines:

"Identifies an associated fragment of an EPUB Content Document.

The value MUST be a [relative-URL-with-fragment string](https://url.spec.whatwg.org/#relative-url-with-fragment-string) [URL] with a non-null fragment identifier."

Any thoughts on this @marisademeglio ?

Reply:
_fragment identifier_ isn't defined in the URL standard. It defines _fragment_ (the component of a URL record) or _URL-fragment string_ (used in the parsing algorithm). If we stay at the string level, the latter should be used. But I'm not sure the term intended to be used outside of the parser algorithm? My gut feeling is that a statements on the URL record would be preferable.

Reply:
> "Identifies an associated fragment of an EPUB Content Document.
> 
> The value MUST be a [relative-URL-with-fragment string](https://url.spec.whatwg.org/#relative-url-with-fragment-string) [URL] with a non-null fragment identifier."
> 
> Any thoughts on this @marisademeglio ?

I think this is very clear. One point though, along the lines of what @rdeltour has been mentioning, "non-null" only appears in the parser algorithm section of the URL spec. I think to express this requirement in URL-speak, we would have to say "a fragment identifier with one or more URL Units". Or could we say "non-empty"? 

Alternately, we could revisit the requirement of having fragment identifiers on MO text references. They're useful, sure, but are they necessary to require all the time? Pointing to the document is enough, if the document is very simple and would only have one sync point anyway. And if you need more, you can use fragment IDs, but you don't have to.

Reply:
I was intentionally trying to avoid the URL standard in this case. 😄 

We have a section on [fragment identifiers for media overlays](https://w3c.github.io/epub-specs/epub33/core/#sec-media-overlays-fragids), so "fragment identifier" is linked to that section for further elaboration.

Does that work?

Reply:
> Alternately, we could revisit the requirement of having fragment identifiers on MO text references.

Do you want to open a new issue for this? Probably not a good idea to get into it in this pull request.

Reply:
Just leaving a note here for my future self: after sketching out a summary of this to open an issue, I've talked myself out of it -- pretty sure the only people whose lives would be made easier would be spec authors ! It introduces a bit of ambiguity that I don't think is worth it.

Reply:
> I was intentionally trying to avoid the URL standard in this case. 😄
> 
> We have a section on [fragment identifiers for media overlays](https://w3c.github.io/epub-specs/epub33/core/#sec-media-overlays-fragids), so "fragment identifier" is linked to that section for further elaboration.
> 
> Does that work?

Yes, so can we omit "non-null", since it seems like all requirements on fragment identifiers then belong in that section 7.3.2.2 ?

As in

"The value MUST be a relative-URL-with-fragment string [URL] with a [fragment identifier](https://w3c.github.io/epub-specs/epub33/core/#sec-media-overlays-fragids)."

I also wonder if the first paragraph of the [fragment identifier section](https://w3c.github.io/epub-specs/epub33/core/#sec-media-overlays-fragids) is necessary. It gives context but it seems to be purely informative.

Reply:
> Yes, so can we omit "non-null"

Works for me!

> I also wonder if the first paragraph of the [fragment identifier section](https://w3c.github.io/epub-specs/epub33/core/#sec-media-overlays-fragids) is necessary.

I'd err on leaving it for the context. I think it would be confusing for anyone entering the section without following one of the links.